tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/phases/OCLFP16SupportPhase.java

```diff
@@ -53,11 +53,11 @@ public Optional<NotApplicable> notApplicableTo(GraphState graphState) {
     protected void run(StructuredGraph graph) {
         boolean fp16Support = false;
         String extensions = null;
-        if (deviceContext.getSPIRVDevice() instanceof OCLDevice) {
-            OCLDevice oclDevice = (OCLDevice) deviceContext.getSPIRVDevice();
+        if (deviceContext.getDevice() instanceof OCLDevice) {
+            OCLDevice oclDevice = (OCLDevice) deviceContext.getDevice();
```

Comment:
i think this can be simplified to avoid casting with the new instance of

Reply:
I agree, but this is not related to this PR. This was because of the refactoring.